### DISCUSSION TITLE
3 general questions about the data and the VITS model

### DISCUSSION THREAD

**Discussion by eyad-al-shami** [coqui-ai/TTS#1041] [General Q&A]
Hi, I am new to the TTS domain, but I have an idea of training an end-to-end model, namely VITS, on the Arabic dataset offered by the Mozilla common-voice project.
I have read the paper "what shall we do with an hour of data?" and from what I understood, the Arabic dataset is big enough to obtain good results.
However, I have some questions **(maybe stupid ones)**:
1. The quality of the recordings is not that good. A lot of them have background noise. How will that affect the quality of the output (maybe if you used the same Mozilla dataset of another language can give me a hint if they also had background noise and how was the result of using it)
2. I started collecting another dataset, but what I have is recordings of length **15 minutes** of audiobooks (total of 13 hours), and from what I read, most of the papers use **2-10** seconds of voice recordings (for Arabic, the average length of the recordings in the Mozilla dataset is **4.16** sec, 263 recordings of length 10 seconds and 3 with 20 sec). My plan is to manually cut the collected recording into smaller ones, but what is the best or acceptable length? and Is my approach of adding more "clean" data is a good idea in the first place? if you can point me to a paper or tell me from your experience what is the relationship between the length of the voices and both the training time and the quality of the output.
3. Lastly, is my choice of using the **VITS** good? Because I saw a lot of questions on using the two-stages models (encoder, vocoder) and not a lot of questions about the VITS, maybe because it is a relatively new model or maybe I misunderstood the model and what it can do.

**Comment by eyad-al-shami** [comment]
Other than these two sections from the Coqui's documentation, which are talking about datasets:

- https://tts.readthedocs.io/en/latest/what_makes_a_good_dataset.html
- https://tts.readthedocs.io/en/latest/formatting_your_dataset.html

I have found two papers that talked about the duration of the voice clips:

- RyanSpeech: A Corpus for Conversational Text-to-Speech Synthesis: [https://arxiv.org/pdf/2106.08468.pdf](https://arxiv.org/pdf/2106.08468.pdf)
- LibriTTS: A Corpus Derived from LibriSpeech for Text-to-Speech: [https://arxiv.org/pdf/1904.02882.pdf](https://arxiv.org/pdf/1904.02882.pdf)

Note: The second paper is really informative and specific.

And a blog post talking about the same thing:
How to create a speech dataset for ASR, TTS, and other speech tasks: [https://ogunlao.github.io/blog/2021/01/26/how-to-create-speech-dataset.html](https://ogunlao.github.io/blog/2021/01/26/how-to-create-speech-dataset.html)

Still did not find anyone talking about training only using the Mozilla Common-voice dataset.
If anyone had tried to do this would you please share the results with us?

**Reply by erogol** [reply]
Common voice is not helpful for TTS since recording quality is unstable and low. But I remember some papers picking some subsets that have reasonable quality.

**Comment by bemnet4u** [comment]
@eyad-al-shami I am curious how your experiment with training a model with just 1 hour went? I tried to do the same to train Tacotron2 model using TensorFlowTTS and after 18k steps I end up getting nan https://github.com/TensorSpeech/TensorFlowTTS/issues/125#issuecomment-1025077851

I am now considering using coqui with as I have seen better support community here.

**Comment by AliMokhammad** [comment]
@eyad-al-shami Thank You for presenting this Subject
I'm also trying to train a custom Arabic model. I'm building my own dataset.
Are there any recommendations about the config.json file params?
I've tried training with "use_phonemes'=True and "phoneme_language" = "ar" but I'm getting errors

**Comment by king-dahmanus** [comment]
Hello there? @eyad-al-shami, if you have a good microphone and the ability to record noise free, Maybe recording the common voice text corpus with your own voice can do the trick. Or, you can use the arabic speech corpus, it's a relatively small one though. I hope this discussion will continue
